### PR TITLE
Unit tests for Pravega sensor collector

### PR DESCRIPTION
**Pull request by ShwethaSNayak**
This pull request introduces new unit tests for the Pravega sensor collector to ensure robust and reliable data collection from sensors.
The tests cover various scenarios, including normal operation, edge cases, and potential error conditions.

Expected:

All tests should pass successfully

### CODE REVIEW

File: pravega-sensor-collector/src/test/resources/LogFileIngestTest.properties [hunk 1]
```diff
@@ -21,3 +21,4 @@ PRAVEGA_SENSOR_COLLECTOR_ACCEL2_SCOPE=examples
 PRAVEGA_SENSOR_COLLECTOR_ACCEL2_CREATE_SCOPE=true
 PRAVEGA_SENSOR_COLLECTOR_ACCEL2_STREAM=sensors-accelerometer
 PRAVEGA_SENSOR_COLLECTOR_ACCEL2_ROUTING_KEY=routingkey1
+PRAVEGA_SENSOR_COLLECTOR_RAW1_MIN_TIME_IN_MILLIS_TO_UPDATE_FILE=5000
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/test/resources/LogFileIngestTest.properties:24]: Please double check the name

File: pravega-sensor-collector/src/test/resources/ParquetFileIngest.properties [hunk 2]
```diff
@@ -20,6 +20,7 @@ PRAVEGA_SENSOR_COLLECTOR_PARQ2_CREATE_SCOPE=false
 PRAVEGA_SENSOR_COLLECTOR_PARQ2_STREAM=stream-p
 PRAVEGA_SENSOR_COLLECTOR_PARQ2_ROUTING_KEY=$(hostname)
 PRAVEGA_SENSOR_COLLECTOR_PARQ2_TRANSACTION_TIMEOUT_MINUTES=2.0
+PRAVEGA_SENSOR_COLLECTOR_RAW1_MIN_TIME_IN_MILLIS_TO_UPDATE_FILE=5000
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/test/resources/ParquetFileIngest.properties:23]: Please double check the name

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/rawfile/RawFileProcessorTests.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+public class RawFileProcessorTests extends FileProcessorTests {
+    @BeforeEach
+    public void before() throws Exception {
+        super.setup();
+
+    }
+    @Test
+    public void generateEventForRawFileTests(){
+        FileProcessor fileProcessor = new RawFileProcessor(config, state, transactionalEventWriter,transactionCoordinator, "test");
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/rawfile/RawFileProcessorTests.java:16]: Do we need any asserts?

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/parquet/ParquetFileProcessorTests.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+public class ParquetFileProcessorTests extends FileProcessorTests {
+
+
+    @BeforeEach
+    public void before() throws Exception {
+        super.setup();
+
+    }
+    @Test
+    public void generateEventForParquetTests(){
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/parquet/ParquetFileProcessorTests.java:17]: Do we need assert.

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/csvfile/CsvFileSequenceProcessorTests.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+public class CsvFileSequenceProcessorTests extends FileProcessorTests {
+
+    @BeforeEach
+    public void before() throws Exception {
+        super.setup();
+
+    }
+    @Test
+    public void generateEventForCSVFileTests(){
+        FileProcessor fileProcessor = new CsvFileSequenceProcessor(config, state, transactionalEventWriter,transactionCoordinator, "test");
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/csvfile/CsvFileSequenceProcessorTests.java:17]: Please add asserts

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/FileProcessorTests.java [hunk 6]
```diff
@@ -91,4 +97,81 @@ public void getEmptyNextFileSet() throws Exception {
...
+    public void processNextFile_CommitException() throws Exception {
+        copyFile();
+        FileProcessor fileProcessor = new RawFileProcessor(config, state, transactionalEventWriter,transactionCoordinator, "test");
+        Mockito.doThrow(TxnFailedException.class).when(transactionalEventWriter).commit();
+        assertThrows(RuntimeException.class, () -> fileProcessor.processFile(new FileNameWithOffset("../../pravega-sensor-collector/parquet-file-sample-data/sub1.parquet", 0), 1L));
+        // Verify that myMethod was called exactly three times
+        Mockito.verify(transactionalEventWriter, Mockito.times(1)).commit();
+    }
+
+    public void copyFile() throws IOException {
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/FileProcessorTests.java:165]: Please add comment

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/FileProcessorTests.java [hunk 7]
```diff
@@ -91,4 +97,81 @@ public void getEmptyNextFileSet() throws Exception {
...
         fileProcessor.processFiles();
     }
+
+    /*
+     * Process the single file for Raw file processor.
+     */
+    @Test
+    public void processNextFile() throws Exception {
+        copyFile();
+       // Mockito.when(state.getNextPendingFileRecord()).thenReturn(new ImmutablePair<>(new FileNameWithOffset("file1.parquet", 0), 1L));
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/FileProcessorTests.java:107]: please remove commented out code

**Review by sachin-j-joshi — CHANGES_REQUESTED** [review]
Minor comments.

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/FileProcessorTests.java (hunk 6 above)
Comment (ShwethaSNayak) [reply pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/FileProcessorTests.java:163]: done

File: pravega-sensor-collector/src/test/resources/LogFileIngestTest.properties (hunk 1 above)
Comment (ShwethaSNayak) [reply pravega-sensor-collector/src/test/resources/LogFileIngestTest.properties]: Done

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/rawfile/RawFileProcessorTests.java (hunk 3 above)
Comment (ShwethaSNayak) [reply pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/file/rawfile/RawFileProcessorTests.java:21]: done

**Review by sachin-j-joshi — APPROVED** [review]
LGTM
